src/main/java/com/commercetools/sync/inventories/helpers/InventoryReferenceResolver.java

```diff
@@ -0,0 +1,177 @@
+     * option is set to true, a new channel will be created with this key and the role {@code "InventorySupply"}.
+     * However, if the {@code ensureChannel} is set to false, the future is completed exceptionally with a
+     * {@link ReferenceResolutionException}.
+     *
+     * @param draft the inventoryEntryDraft to resolve it's channel reference.
+     * @return a {@link CompletionStage} that contains as a result a new inventoryEntryDraft instance with resolved
+     *          supply channel or, in case an error occurs during reference resolution,
+     *          a {@link ReferenceResolutionException}.
+     */
+    @Nonnull
+    public CompletionStage<InventoryEntryDraft> resolveSupplyChannelReference(
+        @Nonnull final InventoryEntryDraft draft) {
+        final Reference<Channel> channelReference = draft.getSupplyChannel();
+        if (channelReference != null) {
+            try {
+                final String keyFromExpansion = getKeyFromExpansion(channelReference);
+                final String channelKey = getKeyFromExpansionOrReference(keyFromExpansion, channelReference);
+                return fetchOrCreateAndResolveReference(draft, channelKey);
+            } catch (ReferenceResolutionException exception) {
+                return CompletableFutureUtils.exceptionallyCompletedFuture(exception);
+            }
+        }
+        return CompletableFuture.completedFuture(InventoryEntryDraftBuilder.of(draft).build());
+    }
+
+    /**
+     * Given an {@link InventoryEntryDraft} and a {@code channelKey} this method fetches the actual id of the
+     * channel corresponding to this key, ideally from a cache. Then it sets this id on the supply channel reference
+     * id of the inventory entry draft. If the id is not found in cache nor the CTP project and {@code ensureChannel}
+     * option is set to true, a new channel will be created with this key and the role {@code "InventorySupply"}.
+     * However, if the {@code ensureChannel} is set to false, the future is completed exceptionally with a
+     * {@link ReferenceResolutionException}.
+     *
+     * @param draft the inventory entry draft to resolve it's supply channel reference.
+     * @param channelKey the key of the channel to resolve it's actual id on the draft.
+     * @return a {@link CompletionStage} that contains as a result a new inventory entry draft instance with resolved
+     *      supply channel reference or an exception.
+     */
+    @Nonnull
+    private CompletionStage<InventoryEntryDraft> fetchOrCreateAndResolveReference(
+        @Nonnull final InventoryEntryDraft draft,
+        @Nonnull final String channelKey) {
+        return channelService.fetchCachedChannelIdByKeyAndRoles(channelKey,
+            Collections.singletonList(ChannelRole.INVENTORY_SUPPLY))
+                             .thenCompose(resolvedChannelIdOptional -> resolvedChannelIdOptional
+                                 .filter(StringUtils::isNotBlank)
```

Comment:
> So there could be a mapping of "someChannelKey" -> null / blankString inside a cache?

No this is not possible

> Isn't an empty Optional returned if the channel doesn't exists?

Yes, you're right and in that case the filter is unnecessary. Will change it ;)